lib/schemas.ts

```diff
@@ -210,9 +227,42 @@ export const experimentFullSchema = experimentBareSchema
     segmentAssignments: yup.array(segmentAssignmentSchema).defined(),
     variations: yup.array<Variation>(variationSchema).defined().min(2),
   })
+  .test(
+    'deployedVariationId-variation-check',
+    'Could not find variation matching deployedVariationId.',
+    /* istanbul ignore next; this is a data validation test itself */
+    (experiment) => {
+      return _.isNumber(experiment.deployedVariationId)
+        ? experiment.variations.some((variation: Variation) => variation.variationId === experiment.deployedVariationId)
+        : true
```

Comment:
I think this would be a bit clearer as
```suggestion
      return _.isNull(experiment.deployedVariation) ||
        experiment.variations.some((variation: Variation) => variation.variationId === experiment.deployedVariationId)
```